src/features/vpn/services.ts

```diff
@@ -16,20 +16,20 @@ const { api } = sbvrUtils;
 const checkAuth = (() => {
 	const authQuery = _.once(() =>
 		api.resin.prepare<{ uuid: string }>({
-			resource: 'device',
-			id: {
-				uuid: { '@': 'uuid' },
-			},
-			options: {
-				$select: 'id',
-			},
+			method: 'POST',
+			url: `device(uuid=@uuid)/canAccess`,
+			body: { action: 'cloudlink' },
 		}),
 	);
 	return multiCacheMemoizee(
 		async (uuid: string, req: permissions.PermissionReq): Promise<number> => {
 			try {
-				const device = await authQuery()({ uuid }, undefined, { req });
-				// for now if the api key is able to read the device then it has vpn access
+				const deviceResult = (await authQuery()({ uuid }, undefined, {
+					req,
+				})) as { d?: Array<{ id: number }> };
+				const device = deviceResult.d?.[0];
+				// Note: /canAccess atm always throws a 401 if it doesn't find a device.
+				// TODO: Rethink whether we need to make a separate request to return a 403.
 				return device != null ? 200 : 403;
```

Comment:
This can just return `200` as `canAccess` either succeeds or throws a 401
```suggestion
				return 200;
```